Intelipro/reja
# CHORE: Refactor codebase to use typescript

#### What does this PR do?
Implements typescript in the codebase
#### Description of Task to be completed?
[x] install typescript and its dependencies
[x] install and setup tslint and its dependencies

#### How should this be manually tested?
- Clone this repo `git clone  [the-repo-link]`
- Checkout to this branch `ch-typescript`
- Install all dependencies `npm install`
- Go through the codebase to see the change and file extenstion

#### Any background context you want to provide?
 It is a strict syntactical superset of JavaScript and adds optional static typing to the language.
TypeScript is a language for application-scale JavaScript.

#### What are the relevant pivotal tracker stories?

#### Screenshots (if appropriate)

#### Questions:
